### ISSUE TITLE
Implementing "designer" keyword

### ISSUE THREAD

**Issue by tajmone** [ascii-boxes/boxes#22]
Hi Thomas, great work! As I pointed out in my pull request, I stepped into an "ethical" problem: I implemented a nice ASCII-Art frame that I've found on the net, one of those that have been goin around for ages. As is often the case for many of these artworks, they have a little signature somewere in the design.
I wasn't able to trace the author (just *akn*), so I've added "akn" it in the boxe sample, specifying that I am just the implementor. I was in doubt as if to put myself as the author or not. I am the implementor, or the author?
I think that boxes project could take a stand on this: I foresee the issue coming up often. It's not nice to just wipe away credits from an author, even if it boils down to 3 letters—but that was exactly how the BBS world was run: nicknames, pseudonyms, ecc.
I've removed the "akn" letters from the box design (they appeared in the lower part), but I did so mainly to save myself the great hassle of having to add another region to the frame. But I made a point of mentioning it in the sample I've created.
Maybe the issue is worth mentioning in the docs, and also you could be implement a tag for "implementor" (or "original designer") to distinguish between the two.
As it is, it is not clear (at least to me) if **author** means the author of the filter, or its designer.
Designer and implementor are more specific. Surely, I have tons of nice ASCII-Art frames and snippets that I could "port" to boxes. Like for cowsay and FIGLet, obviously there is going to be a lot of people fetching old designs from BBS archives and websites, and implement them in boxes. That's how "nostalgics" preserve oldschool and revive it, right? From emulators of old home computers, to ANSI and ASCII Art, MUDs, ecc., there are always waves of revival, but often authorship is lost along the way — still, credit is due where its due!!

**Comment by tsjensen** [comment]
Hi Tristano, thank you, this is a very good point. In fact, looking at the current list of box designs, I see several that would benefit from this extra information. I would propose having two fields then:

- `author` gives the author of the box design, which is the person who created the entry in the boxes config file (which may be a lot of work, too). This would be compatible with the current meaning of the field.
- `designer` gives the name of the artist who created the original artwork.

I would suggest that we also extend the functionality of  the `-l` and `-l -d design` options to include this information. Would you be available for performing the implementation?

**Comment by tajmone** [comment]
I'm afraid C is not my field. I've studied it long time ago, and I wouldn't have a compiler installed on the PC right now. I was thinking more about creating a GUI tools for easying up creation of boxes:
* import a txt file with the frame, and adjust vertical and horizontal rules, to "slice it up" into desired regions (from 1 to 16)
* a form to input author data, revision, config parameters, ecc. I
* save a configuration file to go with the ASCII text-frame, so that settings are kept and boxes can be modified again
* allow testing with a click
* do all sanytize work and export to boxes config file
I was thinking of making in PureBASIC (it would run on Windows, Linux and Mac, both 32 and 64 bits), which offers very nice GUI controls. Also, easy to work with, allow for very fast implementation (maybe a week, or less).
Basically, no need to fiddle with a text editor—I currently use Notepad++, and without RegEx search-&-replace it would take ages to build a complex frame by hand (regex allows me fast selection/deletion from the finished designed, and splicying it into strings). But a GUI tool would be better.

**Comment by tsjensen** [comment]
Don't worry, that's fine. I'll just leave this issue open until someone else (including myself) gets around to doing it. I may add the `designer` keyword to the existing box designs, though.

The authoring tool would be a new and different issue. That would also be really nice to have. When it's complete and stable, we could even host it in the *ascii-boxes* GitHub organization.

**Comment by tsjensen** [comment]
Added `designer` to the [syntax docs](http://boxes.thomasjensen.com/docs/config-syntax.html#name-of-author).

**Comment by tsjensen** [comment]
Done. `-l` and `-l -d` output now honors `designer` information.
